Remove deprecated parts of chisel to get rid of warnings

furkanusta opened the pull request:
Changes:
+ != to =/=
+ Wire(init=x) to WireInit(init=x)
+ RegInit(Vec.fill()) to RegInit(VecInit(Seq.fill()))
+ type.fromBits(data) to data.asTypeOf(type)
+ Decoupled().flip() to Flipped(Decoupled())



There are a few non-chisel (mvc & firrtl) deprecations and pattern matching related warnings but those are left untouched. I might check pattern matching warnings later

Commit:
Remove deprecated parts of chisel to get rid of warnings

Changes:
+ != to =/=
+ Wire(init=x) to WireInit(init=x)
+ RegInit(Vec.fill()) to RegInit(VecInit(Seq.fill()))
+ type.fromBits(data) to data.asTypeOf(type)
+ Decoupled().flip() to Flipped(Decoupled())

aayupov commented:
Thanks, Furkan!

aayupov commented on chisel/dataflow_building_blocks/src/main/scala/VectorAdd.scala:
```diff
@@ -154,13 +154,13 @@ class VecAddMulDP[T <: UInt](gen : T, vecLen : Int)(implicit params : AccParams)
-      val mul3 = Vec.tabulate(vecLen) {i => mul1(i) * add2(i) }
-      val mul4 = Vec.tabulate(vecLen) {i => mul2(i) * add2(i) }
-      val add3 = Vec.tabulate(vecLen) {i => mul3(i) + mul4(i) }
+      val add = VecInit.tabulate(vecLen) {i => io.in1.bits(i) + io.in2.bits(i) }
```
Not sure what this syntax really does and whether it is robust. I usually write it as:
VecInit(Seq.tabulate(vecLen) {...})

aayupov commented on chisel/dataflow_building_blocks/src/test/scala/VectorAddTest.scala:
```diff
@@ -48,7 +48,7 @@ class VecAddTb[T <: UInt](gen : T, vecSize : Int)(implicit params : AccParams) e
   mem.io.mem_wr_out <> va.io.mem_wr_resp 
   
-  va.io.config.bits := va.io.config.bits.fromBits(io.config.asUInt())  
+  va.io.config.bits := io.config.asUInt.asTypeOf(va.io.config.bits)
```
asUInt looks unnecessary

aayupov commented on chisel/designutils/src/main/scala/DecoupledStage.scala:
```diff
@@ -39,7 +39,7 @@ object DecoupledStageInv {
     val q = Module(new DecoupledStage(enq.bits.cloneType))
     q.io.inp.valid := enq.valid // not using <> so that override is allowed
-    q.io.inp.bits := q.io.inp.bits.fromBits(~enq.bits.asUInt)
+    q.io.inp.bits := (~enq.bits.asUInt).asTypeOf(q.io.inp.bits)
```
asUInt is unnecessary

Commit:
Replace VecInit.tabulate() with VecInit(Seq.tabulate())

furkanusta commented:
I've changed VecInit, removed some of the asUInt's (there are couple more but they were used on generic instances where base type was Data, I didn't touch those)  and tried to rebase on top of the master.

Also I've realized that I've added 'unchecked' in one of the matches in ReportTiming.scala while testing, I've reverted that back. In the meantime my editor decided to cleanup all the trailing whitespace in the file and I noticed after pushing that is why it seems like there is a lot of change in ReportTiming.
